Composer/packages/client/src/components/TestController/TestController.tsx

```diff
@@ -114,39 +119,75 @@ export const TestController: React.FC = () => {
     setCalloutVisible(true);
   }
 
-  async function handlePublishLuis(luisConfig) {
+  async function handlePublish(config) {
     setBotStatus(BotStatus.publishing);
     dismissDialog();
-    await setSettings(projectId, { ...settings, luis: luisConfig });
-    await publishLuis(luisConfig, projectId);
+    // save the settings change to store and persist to server
+    const newValue = config;
+    const subscriptionKey = newValue.subscriptionKey;
+    const qnaRegion = newValue.qnaRegion;
+    delete newValue.subscriptionKey;
+    delete newValue.qnaRegion;
+    await setSettings(projectId, {
+      ...settings,
+      luis: newValue,
+      qna: { ...settings.qna, subscriptionKey, region: qnaRegion, endpointKey: '' },
+    });
+    await build(newValue.authoringKey, subscriptionKey, qnaRegion, projectId);
   }
 
   async function handleLoadBot() {
     setBotStatus(BotStatus.reloading);
+    if (settings.qna && Object(settings.qna).subscriptionKey) {
+      await setQnASettings(projectId, Object(settings.qna).subscriptionKey);
+    }
```

Comment:
why used the `Object(value)` initializer?

Reply:
Because setting.qna may is undefined.